lib/cn/cndb.c

```diff
@@ -3284,22 +3275,22 @@ cndb_txn_nak(struct cndb *cndb, u64 txid)
     return ev(err);
 }
 
+#if HSE_MOCKING
 /**
  * cndb_cn_initializer() - Only used in tests
```

Comment:
Seems like this should be moved to the tests then.

Reply:
Agreed, but it's used by different tests, so they would either have to duplicate it or we need some "common" dumping ground for this stuff...

Reply:
We should add an include dir for tests to put shared headers in. tests/unit/cn/include/cn-tests/my_header.h. Thoughts?

Reply:
Agreed, but this might be a trivial case.  Other cases likely need quite a bit of access to internal structures that they wouldn't have if they lived outside the module.  I hate to create a bunch of infrastructure for say, one tiny function :-)

Reply:
Alex had an interesting idea where unit tests would go in the source file directly. What are your thoughts in that department? Would give unit tests easy access to internal symbols.

Reply:
Would need to think about it.  Many of our unit tests have a quite a bit of code, so where would it all go?  Would it all go in one file, or spread around?  If a unit test was restricted to testing one function within the same file then that might work, but many don't do that...
The other problem is most times I really don't want to see all that garbage when I'm searching for stuff.  As it is today I can either include or exclude the tests subdir depending upon what I'm looking for..  If we integrate it then it'll be a mess from a searching standpoint (search/replace in editor could be a pain too).  Also, our unit test code isn't the highest quality, so if we went down this route I'd be inclined to suggest this is a new thing rather than try to pull our existing unit tests directly into the main code base...

Reply:
All in one file.  It would badly bloat the file.  And good point about searching for symbols.  A variation on the approach that solves those two issues is to do something like this at the end of each source file:
```
$ tail -n3 lib/cn/cn.c
#ifdef HSE_UNIT_TEST
#include "unit_test/lib/cn/cn.c"
#endif
```

Reply:
This is a better idea.

Reply:
Agreed,, I was thinking along these same lines, nice to have the test files named for the source file and in a separate directory so that they don't get caught up in my searches! :-)